include/itkSpectra1DAveragingImageFilter.h

```diff
@@ -0,0 +1,120 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkSpectra1DAveragingImageFilter_h
+#define itkSpectra1DAveragingImageFilter_h
+
+#include "itkImageToImageFilter.h"
+#include "itkDefaultConvertPixelTraits.h"
+
+#include <type_traits>
+
+namespace itk
+{
+
+/** \class Spectra1DAveragingImageFilter
+ * \brief Average multiple spectra in input images into a single response line.
+ *
+ *
+ * \ingroup Ultrasound
+ */
+template <typename TInputImage, typename TOutputImage = TInputImage>
+class ITK_TEMPLATE_EXPORT Spectra1DAveragingImageFilter : public ImageToImageFilter<TInputImage, TOutputImage>
```

Comment:
As an aside, is there a pixelwise averaging filter in ITK? Somewhat surprised that there does not seem to be one in ITKImageIntensity. I did find [`itkNaryAddImageFilter`](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Filtering/ImageIntensity/include/itkNaryAddImageFilter.h) just now, but it doesn't seem to offer any advantages over this implementation.

Reply:
Nary filters do not do dimension collapse (3D and 2D to 1D). While this filter could almost certainly be implemented by combining existing filters, I am not sure that would have been simpler. And certainly not more performant.